packages/seacas/libraries/ioss/src/text_mesh/Iotm_TextMeshUtils.h

```diff
@@ -14,593 +14,1754 @@
+  {
+    set_error_handler([](const std::ostringstream &errmsg) { default_error_handler(errmsg); });
+  }
+
+  const std::vector<double> &operator[](const EntityId nodeId) const
+  {
+    auto it(m_nodalCoords.find(nodeId));
+
+    if (it == m_nodalCoords.end()) {
+      std::ostringstream errmsg;
+      errmsg << "Could not find node id " << nodeId;
+      m_errorHandler(errmsg);
+    }
 
-        for (auto iter : m_parts) {
-          names.push_back(iter.second);
-        }
+    return it->second;
+  }
 
-        return names;
-      }
+  void set_coordinate_data(
+      const unsigned spatialDim, const std::set<EntityId> &nodeIds, const std::vector<double> &coordinates)
+  {
+    if (!coordinates.empty()) {
+      validate_num_coordinates(spatialDim, nodeIds, coordinates);
+      fill_coordinate_map(spatialDim, nodeIds, coordinates);
+      m_hasCoordinateData = true;
+    }
+  }
 
-      void set_error_handler(ErrorHandler errorHandler) { m_errorHandler = errorHandler; }
+  void set_error_handler(ErrorHandler errorHandler) { m_errorHandler = errorHandler; }
 
-    private:
-      void handle_block_part(const std::string &name)
-      {
-        const std::string blockPrefix  = "BLOCK_";
-        const unsigned    prefixLength = blockPrefix.length();
+  bool has_coordinate_data() const { return m_hasCoordinateData; }
 
-        if (name.length() < prefixLength + 1)
-          return;
+ private:
+  void validate_num_coordinates(
+      const unsigned spatialDim, const std::set<EntityId> &nodeIds, const std::vector<double> &coordinates)
+  {
+    if (coordinates.size() != nodeIds.size() * spatialDim) {
+      std::ostringstream errmsg;
+      errmsg << "Number of coordinates: " << coordinates.size() << ", Number of nodes: " << nodeIds.size()
+             << ", Spatial dimension: " << spatialDim;
+      m_errorHandler(errmsg);
+    }
+  }
+
+  void fill_coordinate_map(
+      const unsigned spatialDim, const std::set<EntityId> &nodeIds, const std::vector<double> &coordinates)
+  {
+    std::vector<double>::const_iterator coordIter = coordinates.begin();
+    for (const EntityId &nodeId : nodeIds) {
+      m_nodalCoords[nodeId] = std::vector<double>(coordIter, coordIter + spatialDim);
+      coordIter += spatialDim;
+    }
+  }
+
+  bool m_hasCoordinateData{false};
+  std::unordered_map<EntityId, std::vector<double>> m_nodalCoords;
+  ErrorHandler m_errorHandler;
+};
+
+template <typename EntityId, typename Topology>
+struct ElementData {
+  int proc;
+  EntityId identifier;
+  Topology topology;
+  std::vector<EntityId> nodeIds;
+  std::string partName = "";
+
+  operator EntityId() const { return identifier; }
+};
+
+struct SideBlockInfo {
+  std::string name;
+  std::string sideTopology;
+  std::string elementTopology;
+  std::string touchingBlock;
+  std::vector<size_t> sideIndex;
+  unsigned numNodesPerSide;
+};
+
+enum SplitType { SPLIT_BY_TOPOLOGIES = 1, SPLIT_BY_ELEMENT_BLOCK = 2, SPLIT_BY_DONT_SPLIT = 3, SPLIT_INVALID = 4 };
```

Comment:
Should get this from Ioss_SurfaceSplit.h instead of recreating it here.

Reply:
I guess I don't understand why there is a need to have TextMesh be independent of the system.  Why is  a pure STK implementation needed and a pure IOSS implementation and a pure ??? implementation?  It makes the code confusing and duplicates code, but I don't see the benefit.   

The only potential benefit I see is that you can possibly test STK independent of IOSS, but if you are simply replicating code I'm not sure that is a benefit...  In other words, is there truly a need for an independent "SplitType" since that is purely an IOSS construct.  If I grep `SplitType` in the STK code, the only returns are `Ioss::SurfaceSplitType`

Reply:
Yes, the design was to be able to test STK independent of IOSS because we have multiple builds of STK with and without STK_IO. That's one reason TextMesh was set up to be independent of what system it was being used in. The version I mentioned earlier is the STK Mesh concrete implementation of TextMesh which does not go through STK_IO. The TextMesh sideset split code has not been pushed into STK Mesh yet which is why your grep didn't show it. I was working on getting the STK and IOSS versions ready to go at the same time